src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -291,10 +291,12 @@ function escapeQuotesForShell(input: string) {
 
 export function getFeatureLayers(featuresConfig: FeaturesConfig, containerUser: string, remoteUser: string, isBuildah = false, useBuildKitBuildContexts = false, contentSourceRootPath = '/tmp/build-features') {
 
+	const coalesceHomeDir = (user: string) => `$(if [ -d "$(${getEntPasswdShellCommand(user)} | cut -d: -f6)" ]; then echo "$(${getEntPasswdShellCommand(user)} | cut -d: -f6)"; else echo "/home/${user}"; fi)`;
```

Comment:
In terms of the first party Features and it's expectations, this make sense to me. However, from CLI point of view, a user could have a special directory (which is not `/home/$user`) set as it's home directory where this assumption might be conflicting. 

Looping in @chrmarti for his expertise.

Reply:
as long as it's an existing user from the base image being used, then it should resolve to that user's home directory (tested with a private base image)

Reply:
We have avoided making assumptions about the location of the home folder so far. One approach could be to look up the home folder before each feature's install script is run (and again cache it in devcontainer-features.builtin.env).

The problem with having a feature create the remote user remains that `_CONTAINER_USER`, `_CONTAINER_USER_HOME`, `_REMOTE_USER` and `_REMOTE_USER_HOME` are not necessarily valid for the first features that install, only after the feature that create the user has installed can these be valid.

My recommendation is to create the user with a Dockerfile before any features are installed. We have previously discussed doing this automatically (maybe as an opt-in option).

Reply:
Hi @chrmarti , sorry for the delay, I've updated the implementation taking your words into consideration; now both `_CONTAINER_USER_HOME` and `_REMOTE_USER_HOME` are calculated and cached before each feature is installed